### ISSUE TITLE
LamdaLayer and Bottlenect-TransformerLayer 

### ISSUE THREAD

**Issue by innat** [tensorflow/addons#2413]
**Describe the feature and the current behavior/state.**

**LamdaNetwork**: [LambdaNetworks: Modeling long-range Interactions without Attention](https://openreview.net/forum?id=xTJEN-ggl1b)
**BotNet**: [Bottleneck Transformers for Visual Recognition](https://arxiv.org/abs/2101.11605)

For `Lamdanetwork` there are (AFAIK) some open source implementation such: https://github.com/lucidrains/lambda-networks. As it shows promising results, I think it would be good to have some reliable implementation from `tf. addons`.

For `botnet`,  it's officially implemented in `tf 1.x`, [here](https://gist.github.com/aravindsrinivas/56359b79f0ce4449bcb04ab4b56a57a2). As it also shows a great impact on the model performance, it would be great to have it in `tf.keras 2` for ease to use.

**Relevant information**
- Are you willing to contribute it (yes/no): no
- Are you willing to maintain it going forward? (yes/no): no
- Is there a relevant academic paper? (if so, where): yes, mentionedd above
- Is there already an implementation in another framework? (if so, where): yes: botnet's official implementation: https://gist.github.com/aravindsrinivas/56359b79f0ce4449bcb04ab4b56a57a2 
- Was it part of tf. contrib? (if so, where): no, maybe

**Which API type would this fall under (layer, metric, optimizer, etc.)** 
- layer

**Who will benefit from this feature?**
tf. keras user 

**Any other info.**

**Comment by bhack** [comment]
I think that we still don't have enough citations for these.
Please see https://github.com/tensorflow/addons/pull/2370

**Comment by bhack** [comment]
I need to close this please comment this again when It will have more citations.
